02-peer-protocol.md

```diff
@@ -445,6 +828,343 @@ to broadcast the commitment transaction to get his funds back and open a new
+for [`commitment_signed`](#commiting-updates-so-far-commitment_signed) with the following additions.
+
+#### Requirements
+
+The sending node:
+  - MUST send zero HTLC's.
+
+The receiving node:
+  - if the message has one or more HTLC's:
+    - MUST fail the negotiation
+  - if it has not already transmitted its `commitment_signed`:
+    - MUST send `commitment_signed`
+  - Otherwise:
+    - MUST send `tx_signatures`
+
+#### Rationale
+
+The first commitment transaction has no HTLC's.
+
+
+### Sharing funding signatures: `tx_signatures`
+
+After a valid `commitment_signature` has been received
+from the peer and a `commitment_signature` has been sent, a peer:
+  - MUST transmit a [`tx_signatures` message](#the-tx_signatures-message) with their signatures for
+    the funding transaction
+
+
+#### Requirements
+The sending node:
+  - MUST verify it has received a valid commitment signature from its peer
+  - MUST remember the details of this funding transaction
+  - if it has NOT received a valid `commitment_signed` message:
+    - MUST NOT send a `tx_signatures` message
+
+The receiving node:
+  - if the received `witness_stack` weight results in the peer's
+    paid feerate falling below the *opener*'s feerate for the funding
+    transaction:
+    - SHOULD broadcast their commitment transaction, closing the channel.
+  - SHOULD apply `witness`es to the funding transaction and broadcast it
+  - if has already sent or received a `funding_locked` message for this
+    channel:
+    - MUST ignore this message
+
+#### Rationale
+A peer sends their `tx_signatures` as soon as they have received a valid
+`commitments_signed` message.
+
+The channel should be preemptively closed in the case where a peer provides
+valid witness data that causes their paid feerate to fall beneath the
+`open_channel2.funding_feerate_perkw` rate. This penalizes the peer
+for underpayment of fees.
+
+
+### The `init_rbf` Message
+
+This message initiates a replacement of a broadcast funding transaction.
+
+1. type: 72 (`init_rbf`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u64`:`funding_satoshis`]
+   * [`u32`:`locktime`]
+   * [`u32`:`funding_feerate_perkw`]
+
+#### Requirements
+
+The sender:
+  - MUST have sent `open_channel2`
+  - MUST set `funding_feerate_perkw` greater than or equal to
+    65/64 times the last sent `funding_feerate_perkw` rounded down.
+  - MUST NOT have sent or received a `funding_locked` message.
+
+The recipient:
+  - MUST respond with either an error or an `ack_rbf` message.
+  - MUST fail the negotiation if:
+    - the `funding_feerate_perkw` is not greater than 65/64 times
+      `funding_feerate_perkw` of the last successfully negotiated
+      open attempt
+    - they have already sent or received `funding_locked`
+  - MAY fail the negotiation for any reason
+
+#### Rationale
+`funding_feerate_perkw` is the feerate this funding transaction
+will pay. It must be at least 1/64 greater than the last
```

Comment:
Where does 1/64 come from? It will not necessarily make the BIP 125 checks pass (but that could be the responsibility of the peers, to use some greater if needed to pass BIP 125 rules)...

Reply:
It's an arbitrary minimum bar to ensure that your peer doesn't endlessly RBF the channel. 1/64 is 1.5% greater than the last (as a floor); it takes ~45 rounds at most to double the feerate.

We could pick a different number; 1/8th takes about 6 rounds to double.

Note that there's now way to enforce a feerate that guarantees you'll pass the BIP125 checks, as the feerate is communicated prior to the input/outputs for the replacement tx.

Reply:
Got it, I think that I'll use a bigger incremental bump than that in eclair, but I don't mind having this value in the spec (you're right that we just need "some" value).